### PR TITLE
Improve get_oof_pred_proba for refit models

### PR DESCRIPTION
**Pull request by Innixma**
*Issue #, if available:*

*Description of changes:*
- Previously, if `best_model` was not able to predict (aka not saved), calling `get_oof_pred_proba` would error even if the oof_pred_proba was technically available.
  - This PR fixes this.
- Refit models can now be called in `get_oof_pred_proba` without crashing. The `oof_pred_proba` of their parent (original) model will be used as a proxy.
- Code cleanup to maintain parent <-> refit relationship in DAG instead of in a separate dictionary.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by szha** [comment]
Job PR-1833-1 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-1833/1/index.html

**Comment by szha** [comment]
Job PR-1833-2 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-1833/2/index.html

**Comment by Innixma** [comment]
(Note: CI passed for this logic, the error is due to issues outside of this code change

File: core/src/autogluon/core/trainer/abstract_trainer.py [hunk 1]
```diff
@@ -1209,6 +1217,12 @@ def _add_model(self, model: AbstractModel, stack_name: str = 'core', level: int
             del model
         return True
 
+    def _add_model_attr(self, model: str, **attributes):
```
Comment (yinweisu) [inline core/src/autogluon/core/trainer/abstract_trainer.py:1220]: This function name gives people impression that it only add attributes. However, it actually update the attribute. Maybe rename to `_update_model_attr`?

**Review by yinweisu — APPROVED** [review]
LGTM! Added one tiny comment

File: core/src/autogluon/core/trainer/abstract_trainer.py (hunk 1 above)
Comment (Innixma) [reply core/src/autogluon/core/trainer/abstract_trainer.py]: Good catch, updated!

**Pull request merged**